kubernetes/test-infra/eks/node-config/config.tf

```diff
@@ -58,3 +59,169 @@ resource "kubernetes_config_map" "name" {
     )
   }
 }
+
+# This module installs the AWS LoadBalancer Controller
+# https://docs.aws.amazon.com/eks/latest/userguide/alb-ingress.html
+
+data "tls_certificate" "testacc" {
+  url = var.cluster_oidc_issuer_url
+}
+
+resource "aws_iam_openid_connect_provider" "testacc" {
+  client_id_list  = ["sts.amazonaws.com"]
+  thumbprint_list = [data.tls_certificate.testacc.certificates[0].sha1_fingerprint]
+  url             = var.cluster_oidc_issuer_url
+}
+
+data "aws_iam_policy_document" "testacc" {
+  statement {
+    actions = ["sts:AssumeRoleWithWebIdentity"]
+    effect  = "Allow"
+
+    condition {
+      test     = "StringEquals"
+      variable = "${replace(aws_iam_openid_connect_provider.testacc.url, "https://", "")}:sub"
+      values   = ["system:serviceaccount:kube-system:aws-node", "system:serviceaccount:kube-system:aws-load-balancer-controller"]
+    }
+
+    principals {
+      identifiers = [aws_iam_openid_connect_provider.testacc.arn]
+      type        = "Federated"
+    }
+  }
+}
+
+resource "aws_iam_role" "alb" {
+  assume_role_policy = data.aws_iam_policy_document.testacc.json
+  name               = "AWSLoadBalancerControllerIAMPolicy-${var.cluster_name}"
+}
+
+resource "aws_iam_policy" "alb" {
+  name   = "AWSLoadBalancerControllerIAMPolicy-${var.cluster_name}"
+  policy = file("${path.module}/iam_policy.json")
+}
+
+resource "aws_iam_role_policy_attachment" "alb" {
+  policy_arn = aws_iam_policy.alb.arn
+  role       = aws_iam_role.alb.name
+}
+
+resource "kubernetes_service_account" "alb" {
+  depends_on = [var.cluster_name]
+  metadata {
+    name      = "aws-load-balancer-controller"
+    namespace = "kube-system"
+    annotations = {
+      "eks.amazonaws.com/role-arn": aws_iam_role.alb.arn
+    }
+    labels = {
+      "app.kubernetes.io/component": "controller"
+      "app.kubernetes.io/name": "aws-load-balancer-controller"
+    }
+  }
+}
+
+# TODO: use the kubernetes provider to install as much of this as possible
+resource "null_resource" "install-cert-manager-crds" {
```

Comment:
This is kind of an embarrassing hack. The cert-manager stuff comes in one big file:

https://github.com/jetstack/cert-manager/releases/download/v1.0.2/cert-manager.yaml


It will only work if I apply it twice, so that's what these two commands are. Eventually, I'll separate the CRDs from the CRs, etc. There's some dependencies I didn't have time to untangle.